ruddock/modules/rotation/helpers.py

```diff
@@ -88,6 +93,61 @@ def update_votes(prefrosh_id, votes):
     pid=prefrosh_id
   )
 
+def smoothed_average(row):
+    sum_votes = 0
+    num_votes = 0
+
+    for t in VOTE_TUPLES:
+      val = t['vote_value']
+      num = row[t['vote_string']]
+      sum_votes += val * num
+      num_votes += num
+
+    # By adding constants to the numerator and denominator, we smooth out the
+    # distribution of prefrosh ratings, as well as avoid dividing by zero.
+    return (sum_votes + 1.0) / (num_votes + 1.0)
+
+def compute_buckets():
+  query = sqlalchemy.text("""
+    SELECT prefrosh_id,
+           votes_neg_two,
+           votes_neg_one,
+           votes_zero,
+           votes_plus_one,
+           votes_plus_two,
+           votes_plus_three
+    FROM rotation_prefrosh
+  """)
+  results = flask.g.db.execute(query).fetchall()
+
+  query2 = sqlalchemy.text("""
+    SELECT bucket_id, bucket_name
+    FROM rotation_buckets
+  """)
+  buckets = flask.g.db.execute(query2).fetchall()
+  buckets = {r['bucket_id']:float(r['bucket_name']) for r in buckets}
+
+  update = sqlalchemy.text("""
+    UPDATE rotation_prefrosh
+    SET bucket_id=(:bid)
+    WHERE prefrosh_id=(:pid)
+  """)
+  # TODO this _cannot_ be efficient... how can I do a bulk insert in SQL Alchemy?
```

Comment:
I think it would be most efficient to do this entirely within MySQL (either with an update statement for each bucket or a cursor), but at this scale it's not worth your time to change it.

Reply:
yeah, i thought about doing it in mysql, but like you said, it's not at the scale where it matters. it takes like 2 seconds.